src/rez/tests/test_wheel.py

```diff
@@ -0,0 +1,258 @@
+
+    def _execute(self, cmd):
+        assert self.context.execute_shell(command=cmd).wait() == 0
+
+    def _install(self, *packages, **kwargs):
+        return wheel.install(packages, prefix=self.temprepo, **kwargs)
+
+    def _test_install(self, package, version):
+        installed = self._install("%s==%s" % (package, version))
+        assert installed, "Something should have been installed"
+
+        names = [pkg.name for pkg in installed]
+        versions = {
+            package.name: str(package.version)
+            for package in installed
+        }
+
+        self.assertIn(package, names)
+        self.assertEqual(versions[package], version)
+
+    def test_wheel_to_variants1(self):
+        """Test wheel_to_variants with pure-Python wheel"""
+        WHEEL = """\
+Wheel-Version: 1.0
+Generator: bdist_wheel 1.0
+Root-Is-Purelib: true
+Tag: py2-none-any
+Tag: py3-none-any
+"""
+
+        variants = wheel.wheel_to_variants(WHEEL)
+        self.assertEqual(variants, [])
+
+    def test_wheel_to_variants2(self):
+        """Test wheel_to_variants with compiled wheel"""
+        WHEEL = """\
+Wheel-Version: 1.0
+Generator: bdist_wheel 1.0
+Root-Is-Purelib: false
+Tag: cp36-cp36m-win_amd64
+"""
+
+        variants = wheel.wheel_to_variants(WHEEL)
+        self.assertEqual(variants, [
+            "platform-%s" % wheel.platform_name(),
+            "os-%s" % wheel.os_name(),
+            "python-3.6",
+        ])
+
+    def test_wheel_to_variants3(self):
+        """Test wheel_to_variants with unsupported WHEEL"""
+        WHEEL = """\
+Wheel-Version: 2.0
+Generator: bdist_wheel 1.0
+Root-Is-Purelib: false
+Tag: cp36-cp36m-win_amd64
+"""
+
+        self.assertRaises(Exception, wheel.wheel_to_variants, WHEEL)
+
+    def test_wheel_to_variants4(self):
+        """Test wheel_to_variants with pure-Python, solo-version wheel"""
+        WHEEL = """\
+Wheel-Version: 1.0
+Generator: bdist_wheel 1.0
+Root-Is-Purelib: true
+Tag: py2-none-any
+"""
+
+        variants = wheel.wheel_to_variants(WHEEL)
+        self.assertEqual(variants, ["python-2"])
+
+    def test_wheel_to_variants5(self):
+        """Test wheel_to_variants with badly formatted WHEEL"""
+        WHEEL = """\
+I am b'a'd
+"""
+
+        self.assertRaises(Exception, wheel.wheel_to_variants, WHEEL)
+
+    def test_purepython_23(self):
+        """Install a pure-Python package compatible with both Python 2 and 3"""
+        self._test_install("six", "1.12.0")
+
+    def test_purepython_2(self):
+        """Install a pure-Python package only compatible with Python 2"""
+        self._test_install("futures", "3.2.0")
+
+    def test_compiled(self):
+        """Install a compiled Python package"""
+        self._test_install("pyyaml", "5.1")
+
+    # def test_nowheel(self):
+    #     self._install("PySide==1.2.4")
+
+    def test_dependencies(self):
+        """Install mkdocs, which carries lots of dependencies"""
+        installed = self._install("mkdocs==1.0.4")
```

Comment:
You should be careful with this. `mkdocs` sub-depedencies (transitive) could change and the tests will break. Like here: https://github.com/mkdocs/mkdocs/blob/master/setup.py#L59. A new version of click would make the tests fail.

Also, I don't think we should fetch the internet in tests. It's normally error prone and it brings instability to the tests. One could want to develop rez withuot internet. Developers in big studios don't necessarily have access to internet from a shell, or someone could want to write/run some tests while having no internet.

Reply:
Hey jc, thanks for having a look!

> mkdocs sub-depedencies (transitive) could change and the tests will break

Ah, yes that's true. I had assumed that by picking a fixed version of mkdocs, we'd also be getting fixed versions of its dependencies, but that's not the case..

> Also, I don't think we should fetch the internet in tests.

This could probably be managed via the `rez selftest --skip wheel` argument. Alternatively, when `cmake` isn't available on a system, the test is skipped. Maybe the same could apply if the user doesn't have internet.